nilearn/regions/signal_extraction.py

```diff
@@ -15,8 +15,153 @@
 from .. import masking
 from ..image import new_img_like
 
+INF = 1e-9
+
+
+def _check_labels(labels_img, target_shape, target_affine):
+    """Validate shapes and affines of labels.
+    Parameters
+    ----------
+    labels_img : Niimg-like object
+        See http://nilearn.github.io/manipulating_images/input_output.html
+        regions definition as labels. Encodes the region labels of the signals.
+
+    target_shape : numpy.ndarray
+        Desired shape of labels image and mask.
+
+    target_affine : numpy.ndarray
+        Desired affine of labels image and mask.
+
+    Returns
+    -------
+    state : boolean
+        Corresponding state. Is only true for non-empty masks.
+
+    """
+    if labels_img.shape != target_shape:
+        raise ValueError("labels_img and imgs shapes must be identical.")
+    if labels_img.affine.shape != target_affine.shape or \
+       abs(labels_img.affine - target_affine).max() > INF:
+        raise ValueError("labels_img and imgs affines must be identical.")
+
+
+def _check_imgs(target_shape, target_affine, mask_img=None, dim=None):
```

Comment:
Could the naming be more explicit ?

Reply:
Thank you for your helpful feedback @bthirion. What do you think of  `_check_shappe_affine_labels`, `_check_shappe_affine_maps_masks` ?

Reply:
Thank you for your helpful feedback @bthirion. What do you think of  `_check_shappe_affine_labels`, `_check_shappe_affine_maps_masks` ?

Reply:
Yes, there are explicit names.